### PR TITLE
ContractModel: Extend the Escrow contract model

### PR DESCRIPTION
**Pull request by rjmh**
Extend Escrow tests, add negative testing
Add NoLockedFunds to Escrow tests

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

**Comment by rjmh** [comment]
Do you mind having a quick look at this? @sjoerdvisscher

File: plutus-use-cases/test/Spec/Escrow.hs [hunk 1]
```diff
@@ -101,16 +108,20 @@ instance ContractModel EscrowModel where
...
 
   precondition s a = case a of
     WaitUntil slot' -> s ^. currentSlot < slot'
     Redeem _ -> (s ^. contractState . contributions . to fold) `geq` (s ^. contractState . targets . to fold)
              && (s ^. currentSlot < s ^. contractState . refundSlot - 1)
-    Refund w -> s ^. currentSlot > s ^. contractState . refundSlot
+    Refund w -> s ^. currentSlot >= s ^. contractState . refundSlot
              && Nothing /= (s ^. contractState . contributions . at w)
-    Pay w v -> Nothing == s ^. contractState . contributions . at w
+    Pay w v -> (True || Nothing == s ^. contractState . contributions . at w)  -- disabled
```
Comment (sjoerdvisscher) [inline plutus-use-cases/test/Spec/Escrow.hs:120]: Why is this disabled?

Comment (sjoerdvisscher) [reply plutus-use-cases/test/Spec/Escrow.hs:120]: @rjmh

**Review by sjoerdvisscher — APPROVED** [review]

**Comment by MaximilianAlgehed** [comment]
@sjoerdvisscher I think this is ready to merge now :)

**Pull request merged**